src/transformers/models/imagebind/image_processing_imagebind.py

```diff
@@ -0,0 +1,834 @@
+        do_rescale = do_rescale if do_rescale is not None else self.do_rescale
+        rescale_factor = rescale_factor if rescale_factor is not None else self.rescale_factor
+        do_normalize = do_normalize if do_normalize is not None else self.do_normalize
+        image_mean = image_mean if image_mean is not None else self.image_mean
+        image_std = image_std if image_std is not None else self.image_std
+        do_convert_rgb = do_convert_rgb if do_convert_rgb is not None else self.do_convert_rgb
+        do_chunk = do_chunk if do_chunk is not None else self.do_chunk
+        chunk_duration = chunk_duration if chunk_duration is not None else self.chunk_duration
+        num_chunks = num_chunks if num_chunks is not None else self.num_chunks
+        num_frames_per_chunk = num_frames_per_chunk if num_frames_per_chunk is not None else self.num_frames_per_chunk
+        fps = fps if fps is not None else self.fps
+
+        if images is not None:
+            images = make_list_of_images(images)
+        if videos is not None:
+            videos = make_batched_videos(videos)
+
+        validate_kwargs(captured_kwargs=kwargs.keys(), valid_processor_keys=self._valid_processor_keys)
+
+        if (videos is not None and not valid_images(videos)) or (images is not None and not valid_images(images)):
+            raise ValueError(
+                "Invalid input type. Must be of type PIL.Image.Image, numpy.ndarray, "
+                "torch.Tensor, tf.Tensor or jax.ndarray."
+            )
+
+        if images is not None:
+            pixel_values = self._preprocess_image(
+                images=images,
+                do_resize=do_resize,
+                size=size,
+                resample=resample,
+                do_center_crop=do_center_crop,
+                crop_size=crop_size,
+                do_rescale=do_rescale,
+                rescale_factor=rescale_factor,
+                do_normalize=do_normalize,
+                image_mean=image_mean,
+                image_std=image_std,
+                do_convert_rgb=do_convert_rgb,
+                data_format=data_format,
+                input_data_format=input_data_format,
+            )
+        else:
+            pixel_values = []
+            for video in videos:
+                if do_chunk:
+                    clips = self.chunk(
+                        video=video,
+                        fps=fps,
+                        chunk_duration=chunk_duration,
+                        num_chunks=num_chunks,
+                        num_frames_per_chunk=num_frames_per_chunk,
+                    )
+
+                    _pixel_values = [
+                        self._preprocess_image(
+                            images=clip,
+                            is_video=True,
+                            do_resize=do_resize,
+                            size=size,
+                            resample=PILImageResampling.BILINEAR,
```

Comment:
@molbap it's because the default for video is `bilinear` we could either have a specific resampling for video case or  deal with that in the `preprocess` as either images or videos should be passed not both (correct me if I'm wrong @RUFFY-369 ).

We could do something like:

```python
def preprocess(self ,...):
     ....
     if resample is None:
          resample = self.resample if video is None else PILImageResampling.BILINEAR
```

This way we leave the class attribute for the image case (which I suppose will be the most used) and change that accordingly if we're on "video mode".

Reply:
> no I think at this point of the control flow it's indeed always a video? I

Yes, its always a video but `is_video` has a purpose here and the difference is at this [line](https://github.com/huggingface/transformers/blob/f878996feac659a8e93701b4592e2329d54613f9/src/transformers/models/imagebind/image_processing_imagebind.py#L774). Also, please note that only either an image or a video can be provided to the model. `is_video` can be a wrong name for it, it could be `is_clip` as `do_chunk` kwarg decides to perform chunking on a single video. In the original code, chunking or forming clips of a single video was done by default and their resize method is the one to which the control flow goes when`is_video` is True, the `video_resize` function. So, upper half of `if-else` code block deals with chunking and the lower with videos without chunking

Reply:
>  I was thinking of the bilinear resampling as default, if there's a reason for it



> @molbap it's because the default for video is `bilinear` we could either have a specific resampling for video case or deal with that in the `preprocess` as either images or videos should be passed not both (correct me if I'm wrong @RUFFY-369 ).
> 
> We could do something like:
> 
> ```python
> def preprocess(self ,...):
>      ....
>      if resample is None:
>           resample = self.resample if video is None else PILImageResampling.BILINEAR
> ```
> 
> This way we leave the class attribute for the image case (which I suppose will be the most used) and change that accordingly if we're on "video mode".

Yes @EduardoPach is correct. We can either pass an image or a video to  the model as originally designed in the imagebind code itself. `bilinear` mode is used with `do_chunk` `if-else` part or basically the `if` block, is because again, in the original code bilinear interpolation has been done in the transformation part after chunking the video. And the `else` block or when the video which is passed directly without chunking has the default `bicubic`mode which is typically the default mode for many image processing files in transformers or the mode provided by the user.

Reply:
Alright, I understand better. Thanks for explaining!
I think as long as this behaviour is documented it's OK to add it in the `preprocess` method as described by @EduardoPach. We try typically to have inputs and configuration options separated in the control flow but since both processing methods are mixed for image/video, it's not entirely doable. And alright it's in the original code, that should be taken into account in the config indeed! But there might be users that want to choose another setting, so best not to hardcode it

Reply:
> Alright, I understand better. Thanks for explaining! I think as long as this behaviour is documented it's OK to add it in the `preprocess` method as described by @EduardoPach. We try typically to have inputs and configuration options separated in the control flow but since both processing methods are mixed for image/video, it's not entirely doable. And alright it's in the original code, that should be taken into account in the config indeed! But there might be users that want to choose another setting, so best not to hardcode it

@molbap `do_chunk` is specifically for that purpose. The user can give a value for `do_chunk` in the image processor and if `do_chunk` is True, chunking will happen with the bilinear interpolation and `is_video` is just a helper argument for resizing of the chunked video or the clips being passed to the `_preprocess` method